### DISCUSSION TITLE
blog/dynamic-dns-made-easy-with-cloudflare-api/

### DISCUSSION THREAD

**Discussion by giscus** [AkashRajpurohit/AkashRajpurohit#35] [Comments]
# blog/dynamic-dns-made-easy-with-cloudflare-api/

Get rid of the hassle of manual IP updates for your domain with our step-by-step guide on how to create a DDNS using Cloudflare DNS API. Effortlessly keep your domain pointing to your dynamic IP address with just a few simple scripts.

https://akashrajpurohit.com/blog/dynamic-dns-made-easy-with-cloudflare-api/

**Comment by sditit** [comment]
Inspired by your project I wrote my own shell script! Thanks for the inspiration 

https://github.com/sditit/CloudFlareDynamicDNSUpdater

**Reply by AkashRajpurohit** [reply]
This is great!

**Comment by alwynallan** [comment]
Your script was a good start, after 2 days trying to get ddclient to work. I have multiple domains and subdomains pointing to the same IP, so I had to use Zone IDs.

Also, I store the Coludflare IP in a file in /tmp and don't bother making the curl request if the file is there. If the file is not there I write the file for the next time, if the IP changes I overwrite the file. If the server reboots it will get a fresh IP from Cloudflare. Assuming nothing else updates Cloudflare, this will save 1000s of calls to their API.
